receiver/dotnetdiagnosticsreceiver/network/net.go

```diff
@@ -0,0 +1,65 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package network
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"path"
+)
+
+type DialFunc func(network, address string) (net.Conn, error)
+
+type GlobFunc func(pattern string) (matches []string, err error)
+
+// Connect opens an IPC connection to a local dotnet process, given a PID. On
+// Mac/Linux, this is a Unix domain socket. Windows (TBD) will use a named pipe.
+// DialFunc and GlobFunc are swappable for testing.
+func Connect(pid int, dial DialFunc, glob GlobFunc) (net.Conn, error) {
+	tmpdir := os.Getenv("TMPDIR")
+	if tmpdir == "" {
+		tmpdir = "/tmp"
+	}
+	sf, err := socketFile(pid, tmpdir, glob)
+	if err != nil {
+		return nil, err
+	}
+	return dial("unix", sf)
+}
+
+func socketFile(pid int, tempdir string, glob GlobFunc) (string, error) {
```

Comment:
Add comment to explain what this func does.

Reply:
Done.